flepimop/gempyor_pkg/src/gempyor/inference.py

```diff
@@ -316,22 +339,59 @@ def autodetect_scenarios(config):
 
 
 class GempyorInference:
+    """
+    Encapsulates the process of simulation. Provides functionality for parameter
+    inference, SEIR model execution, outcome computation, and inference optimization.
+
+    Attributes:
+        seir_modifiers_scenario:
+        outcome_modifiers_scenario:
+        modinf: A `ModelInfo` object.
+        already_built: Flag to determine if necessary objects have been built.
+        autowrite_seir: Flag to automatically write SEIR data after simulation.
+        static_sim_arguments: Dictionary containing static simulation arguments.
+        do_inference: Flag to determine if model should be run with inference.
+        silent: Flag indicating whether to supress output messaging.
+        save: Flag indicating whether simulation results should be saved.
+    """
+
     def __init__(
         self,
         config_filepath,
         run_id="test_run_id",
-        prefix=None,
-        first_sim_index=1,
-        stoch_traj_flag=False,
+        prefix: str = None,
+        first_sim_index: int = 1,
+        stoch_traj_flag: bool = False,
         rng_seed=None,
-        nslots=1,
-        inference_filename_prefix="",  # usually for {global or chimeric}/{intermediate or final}
-        inference_filepath_suffix="",  # usually for the slot_id
+        nslots: int = 1,
+        inference_filename_prefix: str = "",  # usually for {global or chimeric}/{intermediate or final}
+        inference_filepath_suffix: str = "",  # usually for the slot_id
         out_run_id=None,  # if out_run_id is different from in_run_id, fill this
         out_prefix=None,  # if out_prefix is different from in_prefix, fill this
-        path_prefix="",  # in case the data folder is on another directory
-        autowrite_seir=False,
+        path_prefix: str = "",  # in case the data folder is on another directory
+        autowrite_seir: bool = False,
     ):
+        """
+        Initializes the `GempyorInference` object by loading config, setting
+        up the model, and configuring inference parameters.
+
+        Non-optinal Arg:
+            config_filepath: Path to the config file.
+
+        Optional Args:
+            run_id: ID of the run.
+            prefix: Prefix for files.
+            first_sim_index: Index of first simulation, default is 1.
+            stoch_traj_flag: Whether to run the model stochastically, default is False.
+            rng_seed: Number of slots for parallel computation.
+            nslots: Number of slots, default is 1.
+            inference_filename_prefix: Prefix for inference-related file paths.
+            inference_filepath_suffix: Suffix for inference-related file paths.
+            out_run_id: ID for run output.
+            out_prefix: Prefix for file paths to output.
+            path_prefix: Prefix for paths to files (in case data folder is in another directory)
+            autowrite_seir: Flag to automatically write SEIR data after simulation, defaul is False.
```

Comment:
All of this can go under `Args:`, it is clear from the signature what is and isn't optional.